src/plugins/csp_configuration_provider/server/plugin.ts

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Observable } from 'rxjs';
+import { first } from 'rxjs/operators';
+import {
+  CoreSetup,
+  CoreStart,
+  Logger,
+  OpenSearchClient,
+  Plugin,
+  PluginInitializerContext,
+  SharedGlobalConfig,
+} from '../../../core/server';
+
+import { createCspRulesPreResponseHandler } from './csp_handlers';
+import { OpenSearchCspClient } from './provider';
+import { defineRoutes } from './routes';
+import {
+  CspClient,
+  CspConfigurationProviderPluginSetup,
+  CspConfigurationProviderPluginStart,
+} from './types';
+
+export class CspConfigurationProviderPlugin
+  implements Plugin<CspConfigurationProviderPluginSetup, CspConfigurationProviderPluginStart> {
+  private readonly logger: Logger;
+  private readonly config$: Observable<SharedGlobalConfig>;
+  private cspClient: CspClient | undefined;
+
+  constructor(initializerContext: PluginInitializerContext) {
+    this.logger = initializerContext.logger.get();
+    this.config$ = initializerContext.config.legacy.globalConfig$;
+  }
+
+  private setCspClient(inputCspClient: CspClient) {
+    this.cspClient = inputCspClient;
+  }
+
+  private getCspClient(inputOpenSearchClient: OpenSearchClient) {
+    if (this.cspClient) {
+      return this.cspClient;
+    }
+
+    const openSearchCspClient = new OpenSearchCspClient(inputOpenSearchClient);
+    this.setCspClient(openSearchCspClient);
+    return this.cspClient;
+  }
+
+  public async setup(core: CoreSetup) {
+    this.logger.debug('CspConfigurationProvider: Setup');
```

Comment:
Remove debug code

Reply:
It is automatically generated by the plugin generation helper. Other plugins have this debug statement as well. See advanced settings for example https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/plugins/advanced_settings/server/plugin.ts#L48